### PR TITLE
Add `LocaleFormat` and `TimeZone` format.

### PR DESCRIPTION
**Pull request by jakubmarton**
regarding Locale:
- `Locale#toLanguageTag` and `Locale#fromLanguageTag` do not work for all locales
- I've tried commons-lang `LocaleUtils#toLocale` which did not work for `sr_BA_#Latn` for example.

### CODE REVIEW

File: cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala [hunk 1]
```diff
@@ -177,6 +177,17 @@ trait BaseBsonFormats {
...
 
+  implicit object LocaleBsonFormat extends BasicBsonFormat[Locale] {
+    override def read(b: BsonValue): Locale = {
+      //taken from lift's LocaleField as the simpliest solution, albeit not optimal.
+      //commons-lang LocaleUtils were not able to parse all Locale.getAvailableLocales either.
+      val value = b.asString().getValue
+      Locale.getAvailableLocales.filter(_.toString == value).headOption.getOrElse(defaultValue)
+    }
+    override def write(l: Locale): BsonValue = new BsonString(l.toString)
+    override def defaultValue: Locale = Locale.ENGLISH
```
Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: I think the default we use everywhere is `Locale.US`.

File: cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala [hunk 2]
```diff
@@ -177,6 +177,17 @@ trait BaseBsonFormats {
     override def defaultValue: Currency = Currency.getInstance("USD")
   }
 
+  implicit object LocaleBsonFormat extends BasicBsonFormat[Locale] {
+    override def read(b: BsonValue): Locale = {
+      //taken from lift's LocaleField as the simpliest solution, albeit not optimal.
+      //commons-lang LocaleUtils were not able to parse all Locale.getAvailableLocales either.
+      val value = b.asString().getValue
+      Locale.getAvailableLocales.filter(_.toString == value).headOption.getOrElse(defaultValue)
```
Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: I think this is fine, not sure about letter case thou...?

File: cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala (hunk 1 above)
Comment (jakubmarton) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: I am not sure about it: 
`  val defaultLocale: Locale = new Locale("en")` https://github.com/sgrouples/sgrouples/blob/master/utils/src/main/scala/me/sgrouples/util/Translation.scala#L26

Comment (jakubmarton) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: `static public final Locale ENGLISH = createConstant("en", "");`

Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: Heh, funny, well what I meant was that we user `US` as default everywhere in `rogue-fsqio`. Its funny that its not the same but well.

Comment (mwielocha) [inline cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala]: So the question is: do we want to be consistent with the app or do we want the lib to be inner-consistent ;)

**Pull request merged**